tests/dragonfly/conftest.py

```diff
@@ -334,38 +337,80 @@ def with_ca_tls_client_args(with_tls_client_args, with_tls_ca_cert_args):
     return args
 
 
-def copy_failed_logs_and_clean_tmp_folder(report):
+def copy_failed_logs_and_clean_tmp_folder():
     failed_path = "/tmp/failed"
     path_exists = os.path.exists(failed_path)
     if not path_exists:
         os.makedirs(failed_path)
 
-    if os.path.isfile("/tmp/last_test_log_files.txt"):
+    last_log_file = open("/tmp/failed_list.txt", "r")
+    files = last_log_file.readlines()
+    for file in files:
+        # copy to failed folder
+        shutil.copy(file.rstrip("\n"), failed_path)
+
+    # Clean up everything
+    last_log_file = open("/tmp/failed_list.txt", "w").close()
+    last_log_file = open("/tmp/last_test_log_files.txt", "w").close()
+
+
+def pytest_exception_interact(node, call, report):
+    if report.failed:
+        # To print the test that currently failed
         last_log_file = open("/tmp/last_test_log_files.txt", "r")
+        # Global tracking of all failed tests/logs
+        failed_list = open("/tmp/failed_list.txt", "a")
         files = last_log_file.readlines()
         logging.error(f"Test failed {report.nodeid} with logs: ")
         for file in files:
-            # copy to failed folder
+            failed_list.write(file)
             file = file.rstrip("\n")
             logging.error(f"🪵🪵🪵🪵🪵🪵 {file} 🪵🪵🪵🪵🪵🪵")
-            shutil.copy(file, failed_path)
-
-
-def pytest_exception_interact(node, call, report):
-    if report.failed:
-        copy_failed_logs_and_clean_tmp_folder(report)
-
 
-@pytest.fixture(autouse=True)
+        # Clean it
+        last_log_file = open("/tmp/last_test_log_files.txt", "w").close()
+        global TEST_FAILED
+        TEST_FAILED = True
```

Comment:
so we need a global TEST_FAILED or maybe we can use instead python hook pytest_runtest_makereport

Reply:
`python hook pytest_runtest_makereport` This is per test, `TEST_FAILED` is used at the end of the test session. Would you mind expanding your idea on how `hook pytest_runtest_makereport` would help here?